crates/primitives/src/log.rs

```diff
@@ -0,0 +1,34 @@
+use crate::{Bytes, B256};
+use alloc::vec::Vec;
+
+/// An Ethereum event log object.
+#[derive(Clone, Default, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(
+    feature = "arbitrary",
+    derive(derive_arbitrary::Arbitrary, proptest_derive::Arbitrary)
+)]
+pub struct Log {
+    /// The indexed topic list.
+    pub topics: Vec<B256>,
```

Comment:
`pub` allows for construction of invalid logs with 5+ topics

Reply:
Skill issue

Reply:
types exist to solve skill issues :P

Reply:
i would agree we can add getters instead

Reply:
They might be invalid in EVM, but we can use invalid states for testing and maybe other usecases in the futures. I don't think we should lock ourselves into using arrays since it doesn't really improve anything, might be even worse by bloating the stack size of one log to 160 bytes or more. If you box it well you're back to a vec basically